xc/xenops_server_xen.ml

```diff
@@ -728,7 +728,7 @@ module VM = struct
 		| Reboot -> Domain.Reboot
 		| PowerOff -> Domain.PowerOff
 		| Suspend -> Domain.Suspend
-		| Halt -> Domain.Halt
+		| Halt -> Domain.PowerOff
```

Comment:
`Needs_poweroff` normally means that the domain needs to be destroyed, which happens a few lines down with `VM_destroy`. This is after the guest has cleanly shut itself down. Again a slightly different sense of the word "poweroff" :)